### DISCUSSION TITLE
Testing validity of the data without the actual decompression

### DISCUSSION THREAD

**Discussion by nyurik** [rust-lang/flate2-rs#391] [Q&A]
My [sqlite-compression](https://github.com/nyurik/sqlite-compressions?tab=readme-ov-file#sqlite-compressions) extension needs to support gzip testing in addition to compression/decompression.  Currently I do it with `decode(data).is_ok()`, but that causes a lot of unnecessary memory allocation and possibly other steps.  Is there a way to efficiently validate the content of the gzip stream without allocating/re-allocating output buffers?

**Comment by Byron** [comment]
I don't think there is an official 'validate' call in Gzip, and thus would think decompression is the only way to validate a Gzip stream.
However, it should be possible to decode the stream in chunks and reuse a smaller buffer until done, which then could easily live on the stack. Unfortunately the INFLATE algorithm still allocates a lot during normal operation, making this only a partial fix.
